libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1345,11 +1368,30 @@ bool TypeChecker::visit(Assignment const& _assignment)
 			);
 		// Sequenced assignments of tuples is not valid, make the result a "void" type.
 		_assignment.annotation().type = make_shared<TupleType>();
+
 		expectType(_assignment.rightHandSide(), *tupleType);
 
 		// expectType does not cause fatal errors, so we have to check again here.
-		if (dynamic_cast<TupleType const*>(type(_assignment.rightHandSide()).get()))
+		if (TupleType const* rhsType = dynamic_cast<TupleType const*>(type(_assignment.rightHandSide()).get()))
+		{
 			checkDoubleStorageAssignment(_assignment);
+			// TODO For 0.5.0, this code shoud move to TupleType::isImplicitlyConvertibleTo,
```

Comment:
I think we were using `@todo` in other place (not sure why), but keeping the same makes it easier to find these.